### ISSUE TITLE
Display Mean

### ISSUE THREAD

**Issue by RoyiAvital** [bastibe/Violinplot-Matlab#1]
This is an amazing visualization tool!

Could you add an option to display the Mean of the samples in addition to the Median?
Just adding another marker.

Thank You.

**Comment by RoyiAvital** [comment]
Hi,

I see you update the code frequently.

Could you use Releases for mile stones with release notes?

Thank You.

**Comment by bastibe** [comment]
> Could you add an option to display the Mean of the samples in addition to the Median?

This would go against the philosophy of a violin plot. How would you propose to show the mean? Should it include outliers? I'm genuinely curious.

> I see you update the code frequently.
> Could you use Releases for mile stones with release notes?

So far, I pushed ten commits, most of which were bug fixes. I can certainly tag an arbitrary version, though. Would that be satisfactory?

**Comment by RoyiAvital** [comment]
@bastibe, Interesting question whether it should include outliers or not.
The easy option is to have a parameter to set how to calculate it.

Think of it as an option to display many histograms of distributions.
It would be great to be able to compare, in one plot, as many features of the distribution as possible.
Hence I'd say the mean, by default, should be calculated using all samples.

Regarding the commits and release version.
Well it would be nice sh people will be able to update their function.
Not everybody are doing it using GIT, some just download the ZIP.
So I guess the best option is to add a text file of Release Notes and write there any update made.

Thank You.

**Comment by bastibe** [comment]
I implemented a mean indicator. Could you check if this works for you?

**Comment by RoyiAvital** [comment]
@bastibe ,

You really add release notes or versioning to `README.md` or one of the MATLAB files.
Currently someone who has it already can never know what version he has.

Moreover, in the documentation in `violinplot` suggests that `cats` can not be just regular numeric values while it can (At least the impression is like that).
So I would add a case where `mData` is `m x n` numeric matrix and `vCats` is `n x 1` or `1 x n` numeric vector and a violin is made for each column `i` of `mData` for the `vCats(i)` numeric category.

**Comment by bastibe** [comment]
Would you want to take a stab at that? I would be grateful if you created a pull request with better README!

**Comment by RoyiAvital** [comment]
Hi,
I suggest that you enable Wiki for the repository and then I will be able to add information there.

**Comment by bastibe** [comment]
To be honest, I'd prefer this information in the README. But I'll enable the Wiki you're not comfortable with editing the README.

**Comment by RoyiAvital** [comment]
I don't know the whole Pull Process thing.
It is much easier for me to contribute in Wiki.

**Comment by bastibe** [comment]
> I don't know the whole Pull Process thing.

There should be an edit button in Github when you're looking at the README file (not the project overview, you have to click on the README file once). This should open the README for editing, and package up the result as a Pull Request automatically.

Still, if you prefer the Wiki, I'll pull it into the README afterwards.

**Comment by bastibe** [comment]
At the moment, the Wiki contains only Github's boiler plate. Feel free to add or remove as you see fit.

**Comment by RoyiAvital** [comment]
@bastibe ,
I guess I'm not describing what I mean.

Leave alone the Wiki.
It has nothing to do with versioning.

You need to add information about the version in one of the 3 files in the toolbox (`README.md`, `Violin.m` and `violinplot.m`).
Something to let the user be able to say on what version he works compared to the latest on GIT.

**Comment by bastibe** [comment]
Let's get back on topic. Are you satisfied with the implementation of the mean indicator?

If you want to discuss further topics (adding documentation, adding a changelog, adding versions), please open a new issue.

**Comment by sidpku** [comment]
Maybe the color of mean indicator need to be considered.

![02-violin-Legacy-PFT](https://user-images.githubusercontent.com/37524725/198239194-25ffaedb-73df-4011-b9dc-c3d7843e1a10.png)


if there are too many data points, the mean will be covered.

Maybe Black line?

**Comment by bastibe** [comment]
Good point. We should probably use the same grey tone as all the other lines. Would you like to contribute the change as a pull request?
